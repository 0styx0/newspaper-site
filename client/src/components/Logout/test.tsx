import * as React from 'react';
import LogoutContainer from './container';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import setFakeJwt from '../../tests/jwt.helper';
import { mount } from 'enzyme';
import cache from '../../apolloCache';
import * as sinon from 'sinon';

setFakeJwt({level: 3});

describe('<LogoutContainer />', () => {

    describe('snapshots', () => {

        it('renders correctly', () => {

            const tree = renderer.create(
                <MemoryRouter>
                    <LogoutContainer />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('button#onClick', () => {

        it('removes jwt from localstorage when clicked', () => {

            const wrapper = mount(
                <MemoryRouter>
                    <LogoutContainer />
                </MemoryRouter>
            );

            expect(localStorage.getItem('jwt')).toBeTruthy();

            wrapper.find('button').simulate('click');

            expect(localStorage.getItem('jwt')).toBeFalsy();
        });

        it('clears cache', () => {

            const spy = sinon.spy();
            cache.reset = spy;

            const wrapper = mount(
                <MemoryRouter>
                    <LogoutContainer />
                </MemoryRouter>
            );

            wrapper.find('button').simulate('click');

            expect(spy.called).toBeTruthy();

        });
    });
});
