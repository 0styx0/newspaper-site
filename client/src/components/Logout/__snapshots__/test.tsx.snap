// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutContainer /> snapshots renders correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <button
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
</a>
`;
